### PR TITLE
Development

### PR DESCRIPTION
**Pull request by alexandrabagi**
First attempt to merge development branch to master branch. In the React folder, you can find our React project that we started from Tim's tutorial, we removed the unneccessary files and modified the rest. We have the main screens and a clickable element, in the next iteration we plan to make the rest of the elements clickable and create the rest of the screens.
David's FileMaker files are in the FileMaker folder.

### CODE REVIEW

File: React/src/App.js [hunk 1]
```diff
@@ -43,14 +43,14 @@ class App extends Component {
     window.location = './products/'+ imageToUpdate.id;
   }
 
-  setCategory = category => {
+  /*setCategory = category => {
```
Comment (ayanlee) [inline React/src/App.js]: Please, don't commit commented out code.

File: React/src/App.js [hunk 2]
```diff
@@ -94,12 +94,12 @@ class App extends Component {
               return ( 
                 <HourRegistrationScreen 
                   selectedCategory={this.state.currentCategory}
-                  onCategorySelect={this.setCategory}
+                  //onCategorySelect={this.setCategory}
```
Comment (ayanlee) [inline React/src/App.js]: Remember to delete

File: React/src/screens/HourRegistration.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import React from 'react';
 import InputField from '../components/InputField'
 import Button from './../components/Button'
 
-import Clock from './../images/clock.PNG'
-import { faAlignCenter } from '@fortawesome/free-solid-svg-icons';
+import Clock from './../images/Clock.PNG'
+// import { faAlignCenter } from '@fortawesome/free-solid-svg-icons';
```
Comment (ayanlee) [inline React/src/screens/HourRegistration.js]: Remember to delete

File: React/src/screens/HourRegistration.js [hunk 4]
```diff
@@ -2,8 +2,8 @@ import React from 'react';
...
 
 
 // export const CategoryScreen = props => {
```
Comment (ayanlee) [inline React/src/screens/HourRegistration.js]: Remember to delete

File: React/src/App.js [hunk 5]
```diff
@@ -6,12 +6,12 @@ import {
   Switch
 } from 'react-router-dom';
 
-import { Collections } from './data/Collections';
+// import { Collections } from './data/Collections';
```
Comment (ayanlee) [inline React/src/App.js]: Remember to delete

**Comment by alexandrabagi** [comment]
Dear Mircea, Ayanle,
Here is the summary of the main changes in our commits up to date:
- Implemented popup window onClick for buttons Coworker and Both, we plan to refactor this part once we managed to make all the functionalities work.
- Implemented button design change depending on whether they are selected or not. 
- Deleted commented out lines. 
- Removed unused code for image filters, that we inherited from the original file. 
- Isolated the image screen as an component tha can be reused in both MaterialGridScreen and ProductGridScreen. 
- Added dynamic way of showing different items in each material categoy by using json file.
Our React project files stayed in the React folder, as the project has a separate part which is done in FileMaker, and this structure allows us to unambiguously separate project files of the different parts from each other.
Best,
App Built for Builders team

File: React/src/images/Plastering/data.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+    "images": [
+        {"id": "Plastering/p1", "name": "Type a"},
+<<<<<<< HEAD
```
Comment (ayanlee) [inline React/src/images/Plastering/data.json]: You have to resolve this merge conflict

File: React/src/screens/HourRegistration.js [hunk 7]
```diff
@@ -1,49 +1,95 @@
...
         <div>
           <div style={textStyle}>Who are you registering?</div>
           <div className='row'>
-            <Button className="buttonStyle" label="Me">{this.props.label}</Button>
-            <Button className="buttonStyle" label="Coworker">{this.props.label}</Button>
-            <Button className="buttonStyle" label="Both">{this.props.label}</Button>
-          </div>
-        </div>
+
+            <Button id = {this.state.buttonsRow1[0].id} label={this.state.buttonsRow1[0].label} selected={this.state.buttonsRow1[0].selected} onClick={()=> this.changeButtonState(this.state.buttonsRow1[0].id, 0)}/>
```
Comment (ayanlee) [inline React/src/screens/HourRegistration.js]: Why did you put this in an array, if we can't iterate over this?

File: React/src/screens/ProductsGridScreen.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class ProductGridScreen extends React.Component {
+
+    categoryId = this.props.match.params.categoryId
+    data = require('../images/'+this.categoryId + '/data.json');
+    toggleImageSelect = (id) => {
+      // let imagesToUpdate = [...this.state.images];
+      // let imageToUpdate = imagesToUpdate.find(image => image.id === id);
+      // window.location = './products/'+ imageToUpdate.id;
```
Comment (ayanlee) [inline React/src/screens/ProductsGridScreen.js]: As a general rule, no commented out code :)

**Comment by ayanlee** [comment]
Please fix the merge conflict to get the PR approved :)

File: React/src/images/Plastering/data.json (hunk 6 above)
Comment (alexandrabagi) [reply React/src/images/Plastering/data.json]: The conflict is now resolved :)

**Comment by alexandrabagi** [comment]
Hi! As the merge conflict has been resolved, can we merge Development branch into Master?

**Comment by mircealungu** [comment]
@ayanlee  -- do you agree with the merging? :)

**Comment by ayanlee** [comment]
> @ayanlee -- do you agree with the merging? :)
Yes, it is approved :)

File: React/src/App.js [hunk 9]
```diff
@@ -20,25 +20,31 @@ class App extends Component {
...
+    
   }
 
   toggleImageSelect = (id) => {
-    let imagesToUpdate = [...this.state.images];
-    let imageToUpdate = imagesToUpdate.find(image => image.id === id);
-    window.location = './products/'+ imageToUpdate.id;
+    let materialsToLookUp = [...this.state.materials];
+    let selectedMaterial= materialsToLookUp.find(material => material.id === id);
+    // window.location = './products/'+ selectedMaterial.id +'/' + selectedMaterial.image ;
```
Comment (mircealungu) [inline React/src/App.js]: is the commented out code necessary?

**Pull request merged**